Dockerfile

```
@@ -0,0 +1,8 @@
+FROM rustembedded/cross:aarch64-unknown-linux-gnu
+
+RUN dpkg --add-architecture arm64 && \
+    apt-get update && \
+    apt-get install -t xenial-backports --assume-yes libgtk-3-dev:arm64 x11proto-core-dev:arm64
```

This will pop up anyway some time in the future when someone tries to compile some library that uses `bindgen` to generate bindings to a C library.

This is how I got `bindgen` to work (Compiling v4l in this case): 
```suggestion
    apt-get install -t xenial-backports --assume-yes libgtk-3-dev:arm64 x11proto-core-dev:arm64 && \
    apt-get install -y clang libclang-3.9-dev libc6-dev-i386 gcc-aarch64-linux-gnu
```

---

While I like this idea in theory I wonder if we're better off leaving it out as it doesn't affect the build of this project, either that or add it in but commented out so that someone can add it in when they need bindgen?

---

Leaving it there but commented out might be an idea worth considering. I know it's a pain to try to figure out how to fix the bindgen errors when building with cross. 

The minimum positive outcome from my comment would be that the search engines would pick the comment up and googling "couldn't find any valid shared libraries matching clang.so cross build pinephone" would immediately return this discussion. :smile: 

Leaving it out or adding it in comments is fine by me. At least it's documented here now.

---

Just a thought - do you have a simple sample that uses `bindgen`? I would be more than happy to compile and test it for my phone and then we could build another repo with that built in. Maybe even do a little documentation or a small website/wiki of docs/tips if we can get more people interested?